creator/studies/data_generator/study_generator.py

```diff
@@ -0,0 +1,622 @@
+"""
+Generates fake but realistic Dataservice studies
+Used to setup unit tests for ingest runs and related functionality
+
+Study generator ingest consists of the following steps:
+
+1. (Optional) Delete all previously generated data on disk, then
+   delete study in Dataservice
+
+2. Create a study ingest package at
+<StudyGenerator.working_dir>/<StudyGenerator.study_id>_ingest_package
+
+3. Read existing or create new study data files representing a minimal study
+    - bio_manifest.tsv
+    - sequencing_manifest.tsv
+    - s3_source_gf_manifest.tsv
+    - s3_harmonized_gf_manifest.tsv
+    - gwo_manifest.tsv
+
+4. Load study and sequencing center in Dataservice
+
+5. Ingest study data files into Dataservice using ingest library
+    - Uses the KF ingest lib to do this
+
+Upon completion of ingest, the following entities will  be in Dataservice:
+    - sequencing_center
+    - study
+    - family
+    - participant
+    - biospecimen
+    - genomic_file (only harmonized=False)
+    - biospecimen_genomic_file
+    - sequencing_experiment
+    - sequencing_experiment_genomic_file
+
+"""
+import os
+from collections import defaultdict
+import datetime
+import logging
+import shutil
+import string
+import uuid
+import random as ra
+from pprint import pformat, pprint
+
+from django.conf import settings
+import requests
+import pandas as pd
+
+from d3b_utils.requests_retry import Session
+from kf_lib_data_ingest.common.io import read_df
+from kf_lib_data_ingest.common import pandas_utils
+from kf_lib_data_ingest.app import settings as ingest_settings
+from kf_lib_data_ingest.etl.ingest_pipeline import DataIngestPipeline
+
+from creator.fields import kf_id_generator
+from creator.studies.data_generator.utils import (
+    delete_entities,
+    ENTITY_ENDPOINTS,
+    ENDPOINTS,
+)
+
+# For data generation
+ROOT_DIR = os.path.abspath(os.path.dirname(__file__))
+INGEST_PKG_TEMPLATE = os.path.join(ROOT_DIR, "ingest_package")
+DEFAULT_STUDY_ID = "SD_ME0WME0W"
+BROAD_SC_CENTER = "Broad Institute"
+BROAD_KF_ID = "SC_DGDDQVYR"
+SOURCE_GF_EXTS = [".cram", ".crai", ".cram.md5"]
+HARM_GF_EXT = ".g.vcf.gz"
+DEFAULT_SPECIMENS = 10
+
+
+class StudyGenerator(object):
+    def __init__(
+        self,
+        working_dir=None,
+        dataservice_url=None,
```

Comment:
It does if you don't pass anything to `dataservice_url` (https://github.com/kids-first/kf-api-study-creator/blob/d3db27da3e1636748bba826856cc03abe2aed112/creator/studies/data_generator/study_generator.py#L99)

I kind of want to remove that though and just have the kwarg in the constructor set the url. Since study generator has the ability to delete a study in the dataservice (in its `clean` method), I would rather make a user explicitly pass in the dataservice url so its more obvious which dataservice is being used.